PayPal/src/main/java/com/braintreepayments/api/PayPalClient.java

```diff
@@ -113,12 +129,41 @@ public void onResult(@Nullable final Configuration configuration, @Nullable Exce
                     callback.onResult(manifestInvalidError);
                     return;
                 }
+                String clientId = configuration.getPayPalClientId();
+                if (clientId == null) {
+                    //In the future module, we should default this to our new callback function.
+                    //Is this the correct way to send an analytic?
+                    braintreeClient.sendAnalyticsEvent("paypal.native.client-id-null");
+                    callback.onResult(new BraintreeException("Client Id is null"));
+                    return;
+                }
+                String redirectUrl = getPayPalReturnUrl(activity);
+                Environment environment = Environment.SANDBOX;
+                if ("production".equalsIgnoreCase(configuration.getEnvironment())) {
+                    environment = Environment.LIVE;
+                }
+
+                CheckoutConfig checkoutConfig = new CheckoutConfig(
+                        activity.getApplication(),
+                        clientId,
+                        environment,
+                        redirectUrl
+                );
+                PayPalCheckout.setConfig(checkoutConfig);
                 sendPayPalRequest(activity, payPalCheckoutRequest, callback);
             }
         });
 
     }
 
+    private String getPayPalReturnUrl(Context context) {
+        //Shouldnt the configuration include the return URL of the client? why are generating this?
+        if (context != null) {
+            return String.format("%s://paypalpay", context.getPackageName().toLowerCase(Locale.ROOT)).replace("_", "");
+        }
+        return null;
```

Comment:
I'm deleting underscores to follow Braintree's convention. Question though, why is the return URL generated here? should the return URL come from the `Braintree Configuration`?

Reply:
We should validate this with a sample application that uses underscores as there could be additional work required to make this work. [Here's an issue we're currently working through with the Checkout SDK](https://github.com/paypal/android-checkout-sdk/issues/8) related to underscores in the application id.

Reply:
If the URL is in the Braintree Configuration that would be the best strategy here, but historically we've always done it this way. Might be able to improve our onboarding flow if that return url is present in the config!